pkg/artifactory/resource_artifactory_certificate.go

```diff
@@ -0,0 +1,194 @@
+package artifactory
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"github.com/atlassian/go-artifactory/v2/artifactory"
+	v1 "github.com/atlassian/go-artifactory/v2/artifactory/v1"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArtifactoryCertificate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCertificateCreate,
+		Read:   resourceCertificateRead,
+		Update: resourceCertificateUpdate,
+		Delete: resourceCertificateDelete,
+		Exists: resourceCertificateExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"content": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"fingerprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_to": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"valid_until": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			fingerprint, err := calculateFingerPrint(d.Get("content").(string))
+			if err != nil {
+				return err
+			}
+			if d.Get("fingerprint").(string) != fingerprint {
+				d.SetNewComputed("fingerprint")
```

Comment:
This can cause an error. It should probably be caught, logged and returned. [For example](https://github.com/atlassian/terraform-provider-artifactory/blob/master/pkg/artifactory/resource_artifactory_user.go#L90)

Reply:
I've just checked the error and logged it rather than faff about with a `cascadingErr` for one potential error.

Reply:
LGTM